From 1b0eed6b46543273ecfcd340dc909cddc4567c75 Mon Sep 17 00:00:00 2001
From: Jack Greiner <jack@emoss.org>
Date: Sun, 13 Jul 2025 18:08:36 -0400
Subject: [PATCH] cmake: disable alloca for MinGW builds

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b891873c6..b433cccec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -997,6 +997,9 @@ if(MSVC)
     set(SDL_DISABLE_ALLOCA 1)
   endif()
 endif()
+if(MINGW)
+  set(SDL_DISABLE_ALLOCA 1)
+endof()
 
 # TODO: Can't deactivate on FreeBSD? w/o LIBC, SDL_stdinc.h can't define anything.
 if(SDL_LIBC)
-- 
2.50.1

