From 22d827ecc98154008d25e02af6e7c714524b8948 Mon Sep 17 00:00:00 2001
From: Jack Greiner <jack@emoss.org>
Date: Sun, 13 Jul 2025 17:03:10 -0400
Subject: [PATCH] cmake: Check malloc.h for alloca implementation when building
 with MinGW

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b891873c6..a15652193 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -990,7 +990,7 @@ endif()
 
 set(SDL_DISABLE_ALLOCA 0)
 check_include_file("alloca.h" "HAVE_ALLOCA_H")
-if(MSVC)
+if(MSVC OR MINGW)
   check_include_file("malloc.h" "HAVE_MALLOC_H")
   check_symbol_exists("_alloca" "malloc.h" _ALLOCA_IN_MALLOC_H)
   if(NOT HAVE_ALLOCA_H AND NOT _ALLOCA_IN_MALLOC_H)
-- 
2.50.1

